src/imageops/colorops.rs

```diff
@@ -42,6 +51,28 @@ where
     out
 }
 
+/// Convert the supplied image to a grayscale image with the specified pixel type. Alpha channel is preserved.
+pub fn grayscale_with_type_alpha<NewPixel, I: GenericImageView>(image: &I) -> ImageBuffer<NewPixel, Vec<NewPixel::Subpixel>>
+where
+    Subpixel<I>: 'static,
+    <Subpixel<I> as Num>::FromStrRadixErr: 'static,
```

Comment:
Where does this bound come from? I'm a bit confused where this type even comes into play..

Reply:
Hey, thanks for replying.  I copied the grayscale_with_type method here and changed the to_luma call to a to_luma_alpha

Being as it was copied/modified from the non alpha preserving one it came with the same bounds as grayscale_with_type

I will look into this some more, it might be that we can remove this bound here, and also from the grayscale_with_type function

Reply:
Just want to add, I'm mostly alright with approving the PR in the current state. I'm just very curious. If it turns out to be some convoluted thing it may be better to make an issue for this instead, removing the bound is SemVer compatible.

Reply:
So it seems that this bound was initially introduced in 366b4f57200acc7144914d909484beb7c6985af5, 7 years ago. I imagine this is all related to the rather unfortunate requirements of `Pixel: 'static`. Due to how trait work, this cascades as requirements to all of its associated types and through the `Subpixel: Primitive` and `Primitive: Num` this might be why it was necessary.

Fortunately, the requirement `Pixel: 'static` was recently removed in #1597. It would be interesting to simply remove all these `'static` bounds and see if that compiles.

Reply:
Yeah, it looks like that is precisely the case and it was required to specify the lifetime bounds as a result of those bounds being present on the trait

Since the removal of the `Pixel: 'static` bound in #1597, I was able to remove the lifetime bounds on these functions also